packages/core/src/integration.ts

```diff
@@ -155,3 +164,57 @@ function findIndex<T>(arr: T[], callback: (item: T) => boolean): number {
 
   return -1;
 }
+
+/**
+ * Generate a full integration function from a simple function.
+ * This will ensure to add the given name both to the function definition (as id),
+ * as well as to the integration return value.
+ */
+export function makeIntegrationFn<
```

Comment:
This function is imo not really doing anything. I think we (and eventually our users) are just better off defining integrations as follows.

```ts
export function MyCustomIntegration(): Integration {};
```

Easy, simple, well-typed. The rollup and Vite ecosystem is proof that this pattern works and is well understood.

Reply:
You still need to type all arguments in this case, but I generally agree that this is probably overkill. Will remove this!